x/concentrated-liquidity/simulation/sim_msgs.go

```diff
@@ -68,139 +41,188 @@ func RandMsgCreatePosition(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.
 	if err != nil {
 		return nil, fmt.Errorf("position does not exist")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	var positionIds []uint64
+	for _, position := range positions {
+		positionIds = append(positionIds, position.PositionId)
 	}
 
-	// pick a random position
-	randPosition := positions[rand.Intn(len(positions))]
+	// SwapOwner performs the swap from token0 to token1
+	swapOwner, swapOwnerTokens, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
+	if !senderExists {
+		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
+	}
+
+	// ensure that we always have 2 tokens
+	// Note: tokens returns a random subset of poolDenoms, so  had to add this assertion
+	if len(swapOwnerTokens) < 2 {
+		return nil, fmt.Errorf("user does not have pool tokens")
+	}
+
+	// perform swap until token 1 runs out
+	remainingSwapOwnerToken0Amt := swapOwnerTokens[0].Amount
+	for remainingSwapOwnerToken0Amt.ToDec().TruncateInt().GT(sdk.ZeroInt()) {
+		randToken0Amt := sim.RandomAmount(remainingSwapOwnerToken0Amt)
+
+		if randToken0Amt.LTE(sdk.ZeroInt()) {
+			return nil, fmt.Errorf("invalid amount to swap")
+		}
+
+		// perform swap from token0 to token1 until either token0 or token1 fund runs out
+		_, err = k.SwapExactAmountIn(ctx, swapOwner.Address, poolI, sdk.NewCoin(swapOwnerTokens[0].Denom, randToken0Amt), swapOwnerTokens[1].Denom, sdk.OneInt(), sdk.NewDecWithPrec(1, 2))
+		if err != nil {
+			return nil, err
+		}
+
+		remainingSwapOwnerToken0Amt = remainingSwapOwnerToken0Amt.Sub(randToken0Amt)
+	}
 
 	return &cltypes.MsgCollectFees{
 		Sender:      sender.Address.String(),
-		PositionIds: []uint64{randPosition.PositionId},
+		PositionIds: positionIds,
 	}, nil
 }
 
 func RandMsgCollectIncentives(k clkeeper.Keeper, sim *osmosimtypes.SimCtx, ctx sdk.Context) (*cltypes.MsgCollectIncentives, error) {
-	rand := sim.GetRand()
 	// get random pool
-	_, poolDenoms, err := getRandCLPool(k, sim, ctx)
+	clPool, poolDenoms, err := getRandCLPool(k, sim, ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	// positionCreator creates the position with pool denoms
 	// get random user address with the pool denoms
 	sender, _, senderExists := sim.SelAddrWithDenoms(ctx, poolDenoms)
 	if !senderExists {
 		return nil, fmt.Errorf("no sender with denoms %s exists", poolDenoms)
 	}
 
-	positions, err := k.GetUserPositions(ctx, sender.Address, 0)
+	// get random users position
+	positions, err := k.GetUserPositions(ctx, sender.Address, clPool.GetId())
 	if err != nil {
 		return nil, fmt.Errorf("position does not exist")
 	}
 
-	if len(positions) == 0 {
-		return nil, fmt.Errorf("user does not have any position")
+	if len(positions) < 1 {
+		return nil, fmt.Errorf("user does not have any positions")
 	}
 
-	// pick a random position
-	randPosition := positions[rand.Intn(len(positions))]
+	randPositionId := positions[rand.Intn(len(positions))]
```

Comment:
I think the reason why the failure is non-deterministic is because `math/rand` is used. We should use `rand := sim.GetRand()` to get a properly seeded rand AFAIU